packages/rest/test/acceptance/coercion/coercion.acceptance.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  supertest,
+  createClientForHandler,
+  sinon,
+  SinonSpy,
+} from '@loopback/testlab';
+import {RestApplication, RestServer, get, param} from '../../..';
+
+describe('Coercion', () => {
+  let app: RestApplication;
+  let server: RestServer;
+  let client: supertest.SuperTest<supertest.Test>;
+
+  before(givenAnApplication);
+  before(givenAServer);
+  before(givenAClient);
+
+  after(async () => {
+    await app.stop();
+  });
+
+  class MyController {
+    @get('/create-number-from-path/{num}')
+    createNumberFromPath(@param.path.number('num') num: number) {
+      return num;
+    }
+
+    @get('/create-number-from-query')
+    createNumberFromQuery(@param.query.number('num') num: number) {
+      return num;
+    }
+
+    @get('/create-number-from-header')
+    createNumberFromHeader(@param.header.number('num') num: number) {
+      return num;
+    }
+  }
+
+  it('coerces parameter in path from string to number', async () => {
+    const spy = sinon.spy(MyController.prototype, 'createNumberFromPath');
+    await client.get('/create-number-from-path/100').expect(200);
+    assertCoercion(spy);
```

Comment:
I find the name `assertCoercion` as too cryptic - when reading the test, it's not clear what are we asserting!

One option is to use a more descriptive name, e.g. `assertCalledWithNumber(spy, 100)`.

I find the implementation of the helper as unnecessary long - if the function was called with a number, then surely it could not have been called with a string, because those two scenarios are mutually exclusive.

I am proposing to remove `assertCoercion` entirely and call sinon assertion in each test.

```ts
  it('coerces parameter in path from string to number', async () => {
    const spy = sinon.spy(MyController.prototype, 'createNumberFromPath');
    await client.get('/create-number-from-path/100').expect(200);
    sinon.assert.calledWithExactly(spy, 100);
  });
```

Reply:
done 👍